### PR TITLE
feat(uat): add T22 scenario with steps for large message

### PR DESCRIPTION
**Pull request by auarbekov**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-204
add T22 scenario

**Description of changes:**
- Add T22 scenario
- Add receive step for large message
- Add publish step for large message

**Why is this change necessary:**
Implement scenarios

**How was this change tested:**
Run scenario on CodeBuild

**Test results:**
```
Given my device is registered as a Thing....................................passed
And my device is running Greengrass.........................................passed
When I create a Greengrass deployment with components.......................passed
And I create client device "large_payload_publisher"........................passed
When I associate "large_payload_publisher" with ggc.........................passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.Auth configuration to:.passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.client.Mqtt5JavaSdkClient configuration to:.passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.mqtt.Bridge configuration to:.passed
And I deploy the Greengrass deployment configuration........................passed
Then the Greengrass deployment is COMPLETED on the device after 5 minutes...passed
And the aws.greengrass.clientdevices.mqtt.EMQX log on the device contains the line "is running now!." within 1 minutes.passed
And I discover core device broker as "default_broker" from "large_payload_publisher" in OTF.passed
And I connect device "large_payload_publisher" on aws.greengrass.client.Mqtt5JavaSdkClient to "default_broker" using mqtt "v5".passed
And I subscribe "large_payload_publisher" to "${large_payload_publisher}topic/to/iotcore" with qos 1.passed
When I publish from "large_payload_publisher" to "${large_payload_publisher}topic/to/iotcore" with qos 1 and large message with beginning of "Hello world1" with length 130098.passed
Then message beginning with "Hello world1" and with length 130098 received on "large_payload_publisher" from "${large_payload_publisher}topic/to/iotcore" topic within 10 seconds.passed
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bgklika** [comment]
Why T20 in PR name?  T22 added.

When will be edit PR name please also write T22 not t22

File: uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/api/addon/EventFilter.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ public final class EventFilter {
     private final Integer messageExpiryInterval;
     private final String responseTopic;
     private final byte[] correlationData;
+    private final Boolean isMessageFull;
```
Comment (bgklika) [inline uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/api/addon/EventFilter.java]: Changes in EventFilter is not required for that task.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 2]
```diff
@@ -58,12 +58,14 @@
 import software.amazon.awssdk.utils.CollectionUtils;
 
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: Before we move all details like string modification to lower layers.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 3]
```diff
@@ -58,12 +58,14 @@
...
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Random;
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: Do not use random numbers to create large message. After that you can't ensure you receive exactly the same message as sent !

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 4]
```diff
@@ -917,13 +919,50 @@ private void subscribe(@NonNull String clientDeviceId, @NonNull String topicFilt
         }
     }
 
+    /**
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:920]: Idea of test.

MQTT should deliver exactly the SAME message as was published by another client.
To implement that use below approach.

In step/method for publishing we got beginning of message text as a string and length of whole message. 
On TX side we call method which produce from that short text and length full message. It should not be random. For example we can join text as many times as need to produce full message, or add filler by template.  After that publish_long_message() will just call another publish() method.


When we going to receive the message we also note in the step the  same string and whole message length. Receive step will call the same method to produce predictable full message and just call another receive() method.

**Review by bgklika — CHANGES_REQUESTED** [review]

**Comment by bgklika** [comment]
Please replace "add t20 scenario" with "Add T22 scenario" in body of PR description

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 3 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: deleted in 29f79187174dd5710c02c749df6632e330bc6cf2

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 4 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:920]: fixed in 29f79187174dd5710c02c749df6632e330bc6cf2

File: uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/api/addon/EventFilter.java (hunk 1 above)
Comment (auarbekov) [reply uat/mqtt-client-control/src/main/java/com/aws/greengrass/testing/mqtt/client/control/api/addon/EventFilter.java]: deleted in 29f79187174dd5710c02c749df6632e330bc6cf2

**Review by bgklika — DISMISSED** [review]

**Review by eschastlivtsev — DISMISSED** [review]

**Pull request closed without merging**